support/cas-server-support-shell/src/main/java/org/apereo/cas/shell/commands/EncryptPropertyCommand.java

```diff
@@ -57,14 +61,31 @@ public void encryptValue(
              help = "Key obtention iterations to encrypt",
              optionContext = "Key obtention iterations to encrypt",
              specifiedDefaultValue = StringUtils.EMPTY,
-             unspecifiedDefaultValue = StringUtils.EMPTY) final String iterations) {
+             unspecifiedDefaultValue = StringUtils.EMPTY) final String iterations,
+         @CliOption(key = { "includeBC" },
+             mandatory = false,
+             help = "Include Bouncy Castle provider",
+             specifiedDefaultValue = "true",
+             unspecifiedDefaultValue = "false") final boolean includeBC) {
 
         final CasConfigurationJasyptCipherExecutor cipher = new CasConfigurationJasyptCipherExecutor(this.environment);
         cipher.setAlgorithm(alg);
         cipher.setPassword(password);
+        if (includeBC || BouncyCastleProvider.PROVIDER_NAME.equals(provider)) {
```

Comment:
Sorry to be persistent here; thinking about this again: what is the harm in adding BC by default anyways?

Reply:
Yeah, I was going to mention that the includeBC doesn't really make sense on that command, I can remove includeBC. I think the other commands have to add or remove it based on the option otherwise once it is added it stays added.